source/gui/nvdaControls.py

```diff
@@ -368,49 +368,26 @@ def GetChildRectRelativeToSelf(self, child: wx.Window) -> wx.Rect:
 		When calculating ScrollChildIntoView, the position relative to its parent is not relevant unless the
 		parent is the ScrolledPanel itself. Instead, calculate the position relative to scrolledPanel
 		"""
-		cr = child.GetScreenRect()
-		spr = self.GetScreenPosition()
-		return wx.Rect(cr.x - spr.x, cr.y - spr.y, cr.width, cr.height)
+		childRectRelativeToScreen = child.GetScreenRect()
+		scrolledPanelScreenPosition = self.GetScreenPosition()
+		return wx.Rect(
+			childRectRelativeToScreen.x - scrolledPanelScreenPosition.x,
+			childRectRelativeToScreen.y - scrolledPanelScreenPosition.y,
+			childRectRelativeToScreen.width,
+			childRectRelativeToScreen.height
+		)
 
 	def ScrollChildIntoView(self, child: wx.Window) -> None:
 		"""
-		Uses the same logic as super().ScrollChildIntoView(self, child)
-		except cr = self.GetChildRectRelativeToSelf(child) instead of cr = GetRect()
+		Overrides child.GetRect with `GetChildRectRelativeToSelf` before calling
+		`super().ScrollChildIntoView`. `super().ScrollChildIntoView` incorrectly uses child.GetRect to
+		navigate scrolling, which is relative to the parent, where it should instead be relative to this
+		ScrolledPanel.
 		"""
-		sppu_x, sppu_y = self.GetScrollPixelsPerUnit()
-		vs_x, vs_y = self.GetViewStart()
-		cr = self.GetChildRectRelativeToSelf(child)
-		clntsz = self.GetClientSize()
-		new_vs_x, new_vs_y = -1, -1
-
-		# is it before the left edge?
-		if cr.x < 0 and sppu_x > 0:
-			new_vs_x = vs_x + (cr.x / sppu_x)
-
-		# is it above the top?
-		if cr.y < 0 and sppu_y > 0:
-			new_vs_y = vs_y + (cr.y / sppu_y)
-
-		# For the right and bottom edges, scroll enough to show the
-		# whole control if possible, but if not just scroll such that
-		# the top/left edges are still visible
-
-		# is it past the right edge ?
-		if cr.right > clntsz.width and sppu_x > 0:
-			diff = math.ceil(1.0 * (cr.right - clntsz.width + 1) / sppu_x)
-			if cr.x - diff * sppu_x > 0:
-				new_vs_x = vs_x + diff
-			else:
-				new_vs_x = vs_x + (cr.x / sppu_x)
-
-		# is it below the bottom ?
-		if cr.bottom > clntsz.height and sppu_y > 0:
-			diff = math.ceil(1.0 * (cr.bottom - clntsz.height + 1) / sppu_y)
-			if cr.y - diff * sppu_y > 0:
-				new_vs_y = vs_y + diff
-			else:
-				new_vs_y = vs_y + (cr.y / sppu_y)
-
-		# if we need to adjust
-		if new_vs_x != -1 or new_vs_y != -1:
-			self.Scroll(new_vs_x, new_vs_y)
+		oldChildGetRectFunction = child.GetRect
+		child.GetRect = lambda: self.GetChildRectRelativeToSelf(child)
+		try:
+			super().ScrollChildIntoView(child)
+		finally:
+			# ensure child.GetRect is reset properly even if super().ScrollChildIntoView throws an exception
+			child.GetRect = oldChildGetRectFunction

```

Comment:
I avoided suggesting this because it's a little dangerous. If any of the other method calls within `super().ScrollChildIntoView` rely on the absolute value for rect for the child, this will be broken in a subtle way. 
Worse, by the time we update wxPython, we may have forgotten about this.

Reply:
`child` is only used once in `super().ScrollChildIntoView`, and that is to calculate the incorrect `child.GetRect` https://github.com/wxWidgets/Phoenix/blob/wxPython-4.1.1/wx/lib/scrolledpanel.py#L190. Do we only update `wxPython` with compatibility breaking releases? That way I can mark this as deprecated for 2022.1 so it won't get lost.

Reply:
I think it makes sense that we only update on compat breaking releases. Let's mark it as deprecated.